test/rewards/rewards.test.ts

```diff
@@ -594,13 +601,71 @@ describe('Rewards', () => {
         // After state
         const afterTokenSupply = await grt.totalSupply()
         const afterIndexer1Stake = await staking.getIndexerStakedTokens(indexer1.address)
+        const afterIndexer1Balance = await grt.balanceOf(indexer1.address)
+        const afterStakingBalance = await grt.balanceOf(staking.address)
 
         // Check that rewards are put into indexer stake
         // NOTE: calculated manually on a spreadsheet
         const expectedIndexerStake = beforeIndexer1Stake.add(expectedIndexingRewards)
         const expectedTokenSupply = beforeTokenSupply.add(expectedIndexingRewards)
-        // Check
+        // Check stake should have increased with the rewards staked
+        expect(toRound(afterIndexer1Stake)).eq(toRound(expectedIndexerStake))
+        // Check indexer balance remains the same
+        expect(afterIndexer1Balance).eq(beforeIndexer1Balance)
+        // Check indexing rewards are kept in the staking contract
+        expect(toRound(afterStakingBalance)).eq(
+          toRound(beforeStakingBalance.add(expectedIndexingRewards)),
+        )
+        // Check that tokens have been minted
+        expect(toRound(afterTokenSupply)).eq(toRound(expectedTokenSupply))
+      })
+
+      it('should distribute rewards on closed allocation and send to rewards pool', async function () {
+        await staking.connect(indexer1.signer).setRewardsDestination(RewardsDestination.RewardsPool)
+
+        // Setup
+        await setupIndexerAllocation()
+
+        // Jump
+        await advanceBlocks(await epochManager.epochLength())
+
+        // Before state
+        const beforeTokenSupply = await grt.totalSupply()
+        const beforeIndexer1Stake = await staking.getIndexerStakedTokens(indexer1.address)
+        const beforeIndexer1Balance = await grt.balanceOf(indexer1.address)
+        const beforeStakingBalance = await grt.balanceOf(staking.address)
+
+        const expectedIndexingRewards = toGRT('1471954234')
```

Comment:
this was calculated on a spread sheet?

Reply:
Originally we calculated all rewards tests agains a spreedsheet with the formulas, based on issuance rate and blocks, we could make the spreadsheet public and link